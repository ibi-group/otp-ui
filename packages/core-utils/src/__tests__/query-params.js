import queryParams, { getCustomQueryParams } from "../query-params";

describe("query-params", () => {
  describe("getCustomQueryParams", () => {
    it("should return queryParams itself if no customizations", () => {
      expect(getCustomQueryParams()).toEqual(queryParams);
    });

    it("should return queryParams with customizations", () => {
      const customizations = {
        maxWalkDistance: "Max Walk Distance In Meters",
        "maxWalkDistance.options": [
          {
            text: "200 m",
            value: 100
          },
          {
            text: "500 m",
            value: 500
          }
        ]
      };
      const customizedQueryParams = getCustomQueryParams(customizations);
      expect(customizedQueryParams).toMatchSnapshot();
    });
  });
});
