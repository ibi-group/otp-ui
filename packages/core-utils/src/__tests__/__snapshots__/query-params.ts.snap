// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query-params getCustomQueryParams should return queryParams with customizations 1`] = `
Array [
  Object {
    "default": null,
    "itineraryRewrite": [Function],
    "name": "from",
  },
  Object {
    "default": null,
    "itineraryRewrite": [Function],
    "name": "to",
  },
  Object {
    "default": [Function],
    "name": "date",
  },
  Object {
    "default": [Function],
    "name": "time",
  },
  Object {
    "default": "NOW",
    "itineraryRewrite": [Function],
    "name": "departArrive",
  },
  Object {
    "default": "WALK,TRANSIT",
    "name": "mode",
  },
  Object {
    "name": "companies",
  },
  Object {
    "name": "bannedRoutes",
  },
  Object {
    "name": "preferredRoutes",
  },
]
`;
